apk-generator/index.html

```diff
@@ -121,13 +121,19 @@ <h3 id="status"></h3>
                 $('#generator-progress').css('display', 'block')
                 $('#upload-progress').css('display', 'block')
                 var ary = $(form).serializeArray();
+                var emailse = $("#Email").val();
                 var obj = {};
                 updatePercentUpload(10);
                 for (var a = 0; a < ary.length; a++) obj[ary[a].name] = ary[a].value;
                 console.log("JSON", obj);
                 if (obj.Email == "" || obj.App_Name == "") {
```

Comment:
Yes do it out remove the `setTimeout("location.reload(true);", 1);

Plus what if I show all errors in a alert box. That will be a good UX right. @the-dagger .

The code now does is that first it checks whether email and api-links are empty if empty the user is asked to correct it. Then it checks for email to be right, if not it will alert and then proceed. Two alerts will never be shown at the same time as "if and else if" are used. If only "if and if" are used it will throw up 2 alerts...